src/App.php

```diff
@@ -156,18 +166,40 @@ public function processRequest(ServerRequestInterface $request, float $start_tim
 		);
 	}
 
-	private function setupContainerForRunningFrontend(ServerRequestInterface $request): void
+	public function processEjabberd(): void
+	{
+		$this->setupContainerForAddons();
+
+		$this->container = $this->container->addRule(LoggerInterface::class,[
+			'constructParams' => [LogChannel::AUTH_JABBERED],
+		]);
+
+		$this->setupLegacyServerLocator();
+
+		$this->registerErrorHandler();
+
+		// Check the database structure and possibly fixes it
+		\Friendica\Core\Update::check(\Friendica\DI::basePath(), true);
+
+		$appMode = $this->container->create(Mode::class);
+
+		if ($appMode->isNormal()) {
+			/** @var ExAuth $oAuth */
+			$oAuth = $this->container->create(ExAuth::class);
+			$oAuth->readStdin();
+		}
+	}
+
+	private function setupContainerForAddons(): void
 	{
 		/** @var \Friendica\Core\Addon\Capability\ICanLoadAddons $addonLoader */
 		$addonLoader = $this->container->create(\Friendica\Core\Addon\Capability\ICanLoadAddons::class);
 
 		$this->container = $this->container->addRules($addonLoader->getActiveAddonConfig('dependencies'));
-		$this->container = $this->container->addRule(Mode::class, [
-			'call' => [
-				['determineRunMode', [false, $request->getServerParams()], Dice::CHAIN_CALL],
-			],
-		]);
+	}
 
+	private function setupLegacyServerLocator(): void
```

Comment:
Shouldn't be this named `setupLegacyServiceLocator`, related to the conversation we've had recently about the Service Locator anti-pattern?
```suggestion
	private function setupLegacyServiceLocator(): void
```

Reply:
You are right, fixed.